README.md

```diff
@@ -588,7 +588,33 @@ Expressions can contain property placeholders, such as `#{${max.delay}}` or
 - `exceptionExpression` is evaluated against the thrown exception as the `#root` object.
 - `maxAttemptsExpression` and the `@BackOff` expression attributes are evaluated once,
 during initialization. There is no root object for the evaluation but they can reference
-other beans in the context.
+other beans in the context
+
+Starting with version 2.0, expressions in `@Retryable`, `@CircuitBreaker`, and `BackOff` can be evaluated once, during application initialization, or at runtime.
+With earlier versions, evaluation was always performed during initialization (except for `Retryable.exceptionExpression` which is always evaluated at runtime).
+When evaluating at runtime, a root object containing the method arguments is passed to the evaluation context.
+
+**Note:** The arguments are not available until the method has been called at least once; they will be null initially, which means, for example, you can't set the initial `maxAttempts` using an argument value, you can, however, change the `maxAttempts` after the first failure and before any retries are performed.
```

Comment:
Is it only a limitation for the `maxAttempts` option?

I see `open()` is done far before a `doWithRetry()`, so perhaps those `args` does not make sense for circuit breaker...

Looks like BackOff expressions don't suffer from this limitation: it is indeed called in the `catch` from the `doWithRetry()`.

Do I miss anything?

Thanks

Reply:
It turns out that we can't use args at all for any stateful retry (including circuit breakers) because you would get the args from the previous call (which may be different).